src/library/scala/util/Using.scala

```diff
@@ -44,20 +44,53 @@ import scala.util.control.{ControlThrowable, NonFatal}
   * import java.io.{BufferedReader, FileReader}
   * import scala.util.{Try, Using}
   *
+  * val files = List("file1.txt", "file2.txt", "file3.txt", "file4.txt")
   * val lines: Try[Seq[String]] = Using.Manager { use =>
-  *   val r1 = use(new BufferedReader(new FileReader("file1.txt")))
-  *   val r2 = use(new BufferedReader(new FileReader("file2.txt")))
-  *   val r3 = use(new BufferedReader(new FileReader("file3.txt")))
-  *   val r4 = use(new BufferedReader(new FileReader("file4.txt")))
+  *   // acquire resources
+  *   def reader(filename: String) = use(new BufferedReader(new FileReader(filename)))
   *
   *   // use your resources here
   *   def lines(reader: BufferedReader): Iterator[String] =
   *     Iterator.continually(reader.readLine()).takeWhile(_ != null)
   *
-  *   (lines(r1) ++ lines(r2) ++ lines(r3) ++ lines(r4)).toList
+  *   files.map(reader).flatMap(lines)
   * }
   * }}}
   *
+  * Composed or "wrapped" resources may be acquired in order of construction,
+  * if "underlying" resources are not closed:
+  * {{{
+  *   def reader(filename: String) = use(new BufferedReader(use(new FileReader(filename))))
```

Comment:
the documentation is a bit unclear (and I'm not sure if it actually specifies the behaviour), but in practice, `BufferedReader` closes the `Reader` it wraps when you call `close()`. I'm pretty sure I just forgot to wrap the `FileReader`s when I originally wrote my example and did not actually know that at the time though

Reply:
I almost added, "if closing twice is benign". I think anyone who uses the ancient Java idiom has wondered whether it's necessary to close the "inner" thing. (It isn't, no one does.) I thought this was a non-threatening way to introduce the idea.